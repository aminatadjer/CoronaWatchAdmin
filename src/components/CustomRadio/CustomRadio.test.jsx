import React from 'react';
import ReactDom from 'react-dom';
import CustomRadio from './CustomRadio';
import { render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDom.render(<CustomRadio></CustomRadio>, div);
});

// check props
it('renders CustomButton correctly', ()=>{
    const {getByTestId} = render(<CustomRadio option={true}></CustomRadio>)
    expect(getByTestId('customrd')).toBeChecked;
});

it('renders CustomButton correctly', ()=>{
    const {getByTestId} = render(<CustomRadio label="This is a radiobutton"></CustomRadio>)
    expect(getByTestId('customrd')).toHaveTextContent("This is a radiobutton");
});

//snapshot

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomRadio label="This is a radiobutton"></CustomRadio>).toJSON();
    expect(tree).toMatchSnapshot();
})

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomRadio number={1}></CustomRadio>).toJSON();
    expect(tree).toMatchSnapshot();
})

it ('matches snapshot', ()=>{
    const tree = renderer.create(<CustomRadio option ={true}></CustomRadio>).toJSON();
    //const tree = renderer.create(<CustomRadio option={true}></CustomRadio>).toJSON();
    expect(tree).toMatchSnapshot();
})