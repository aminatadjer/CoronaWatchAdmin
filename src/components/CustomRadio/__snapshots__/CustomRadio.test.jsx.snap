// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="radio"
  data-testid="customrd"
>
  <input
    type="radio"
  />
  <label>
    This is a radiobutton
  </label>
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="radio"
  data-testid="customrd"
>
  <input
    id={1}
    type="radio"
  />
  <label
    htmlFor={1}
  />
</div>
`;

exports[`matches snapshot 3`] = `
<div
  className="radio"
  data-testid="customrd"
>
  <input
    type="radio"
    value={true}
  />
  <label />
</div>
`;
